lib/chef/rest.rb

```diff
@@ -39,6 +39,152 @@ class REST < HTTP; end
 require 'chef/http/remote_request_id'
 
 class Chef
+
+  class SocketlessChefZeroClient
+
+    module Response
+
+      def read_body(dest = nil, &block)
```

Comment:
I should document why this is necessary. tl;dr, we set `@read` to true, but `Net::HTTP` won't let you do a streaming style body handler (with a block) after reading the body the first time.